designer/server/src/test/scala/pl/touk/nussknacker/ui/api/NodesApiSpec.scala

```diff
@@ -0,0 +1,1093 @@
+package pl.touk.nussknacker.ui.api
+
+import io.restassured.RestAssured.given
+import io.restassured.module.scala.RestAssuredSupport.AddThenToResponse
+import org.hamcrest.Matchers.equalTo
+import org.scalatest.freespec.AnyFreeSpecLike
+import pl.touk.nussknacker.engine.api.process.ProcessName
+import pl.touk.nussknacker.engine.build.ScenarioBuilder
+import pl.touk.nussknacker.engine.canonicalgraph.CanonicalProcess
+import pl.touk.nussknacker.test.{
+  NuRestAssureExtensions,
+  NuRestAssureMatchers,
+  PatientScalaFutures,
+  RestAssuredVerboseLogging
+}
+import pl.touk.nussknacker.ui.api.helpers._
+
+class NodesApiSpec
+    extends AnyFreeSpecLike
+    with NuItTest
+    with WithMockableDeploymentManager
+    with NuScenarioConfigurationHelper
+    with NuRestAssureExtensions
+    with NuRestAssureMatchers
+    with RestAssuredVerboseLogging
+    with PatientScalaFutures {
+
+  val processName: ProcessName = ProcessName("test")
+
+  val process: CanonicalProcess = ScenarioBuilder
+    .streaming(processName.value)
+    .source("sourceId", "barSource")
+    .emptySink("sinkId", "barSink")
+
+  "The endpoint for nodes" - {
+    "additional info when" - {
+      "authenticated should" - {
+        "return additional info for existing process" in {
+          createSavedProcess(process, TestCategories.Category1, TestProcessingTypes.Streaming)
+
+          given()
+            .basicAuth("allpermuser", "allpermuser")
+            .and()
+            .jsonBody(
+              s"""{
+                 |    "id": "enricher",
+                 |    "service": {
+                 |        "id": "paramService",
+                 |        "parameters": [
+                 |            {
+                 |                "name": "id",
+                 |                "expression": {
+                 |                    "language": "spel",
+                 |                    "expression": "'a'"
+                 |                }
+                 |            }
+                 |        ]
+                 |    },
+                 |    "output": "out",
+                 |    "additionalFields": null,
+                 |    "type": "Enricher"
+                 |}""".stripMargin
+            )
+            .when()
+            .post(s"$nuDesignerHttpAddress/api/nodes/${process.name}/additionalInfo")
+            .Then()
+            .statusCode(200)
+            .body(
+              equalsJson(s"""{
+                   |  "content": "\\nSamples:\\n\\n| id  | value |\\n| --- | ----- |\\n| a   | generated |\\n| b   | not existent |\\n\\nResults for a can be found [here](http://touk.pl?id=a)\\n",
+                   |  "type": "MarkdownAdditionalInfo"
+                   |}""".stripMargin)
+            )
+
+          given()
```

Comment:
I wonder if we shouldn't have two tests instead of this one? At the moment it's hard to understand why there are two different responses and how the two requests differs (from the business point of view)